File: pandora-core/src/main/java/tech/linjiang/pandora/ExpandController.java
package tech.linjiang.pandora;

import java.util.ArrayList;
import java.util.List;

import tech.linjiang.pandora.expand.ExpandItem;
import tech.linjiang.pandora.expand.ExpandType;

public class ExpandController {

    private final List<ExpandItem> expandItemList = new ArrayList<>();

    public void addExpandItem(ExpandItem expandItem){
        if(!expandItemList.contains(expandItem)){
            expandItemList.add(expandItem);
        }
    }

    public List<ExpandItem> getExpandItem(ExpandType expandType){
        List<ExpandItem> expandList = new ArrayList<>();
        for (ExpandItem expandItem : expandItemList) {
            if(expandItem.getExpandType() == expandType){
                expandList.add(expandItem);
            }
        }
        return expandList;
    }
}

File: pandora-core/src/main/java/tech/linjiang/pandora/ExpandController.kt
package tech.linjiang.pandora

import tech.linjiang.pandora.expand.ExpandItem
import tech.linjiang.pandora.expand.ExpandType

class ExpandController {

    private val expandItemList = mutableListOf<ExpandItem>()

    fun addExpandItem(expandItem: ExpandItem){
        if(!expandItemList.contains(expandItem)){
            expandItemList.add(expandItem)
        }
    }

    fun getExpandItem(expandType: ExpandType):List<ExpandItem>{
        val expandList = mutableListOf<ExpandItem>()
        expandItemList.forEach {
            if(it.expandType == expandType){
                expandList.add(it)
            }
        }
        return expandList
    }
}